sample-authn/src/main/java/net/trajano/ms/example/authn/AuthnResource.java

```diff
@@ -64,6 +66,11 @@ public OAuthTokenResponse json(@FormParam("j_username") final String username,
         }
         final JwtClaims claims = new JwtClaims();
         claims.setSubject(username);
+        try {
+            claims.setAudience(HttpAuthorizationHeaders.parseBasicAuthorization(authorization)[0]);
+        } catch (final ParseException e) {
+            throw OAuthTokenResponse.unauthorized(ErrorCodes.UNAUTHORIZED_CLIENT, "Invalid or missing authorization", String.format("Basic realm=\"%s\", encoding=\"UTF-8\"", "authz"));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)